tests/v2/validation/provisioning/k3s/post_kdm_oob_release_test.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type KdmChecksTestSuite struct {
+	suite.Suite
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+}
+
+const (
+	defaultNamespace = "default"
+)
+
+func (k *KdmChecksTestSuite) TearDownSuite() {
+	k.session.Cleanup()
+}
+
+func (k *KdmChecksTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	k.session = testSession
+
+	k.ns = defaultNamespace
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	k.k3skubernetesVersions = clustersConfig.K3SKubernetesVersions
+
+	k.cnis = clustersConfig.CNIs
+	k.providers = clustersConfig.Providers
+	k.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(k.T(), err)
+
+	k.client = client
+}
+
+func (k *KdmChecksTestSuite) TestK3SK8sVersions() {
+	logrus.Infof("checking for valid k8s versions..")
+	require.GreaterOrEqual(k.T(), len(k.k3skubernetesVersions), 1)
+	// fetching all available k8s versions from rancher
+	releasedK8sVersions, _ := versions.ListK3SAllVersions(k.client)
+	logrus.Info("expected k8s versions : ", k.k3skubernetesVersions)
+	logrus.Info("k8s versions available on rancher server : ", releasedK8sVersions)
+	for _, expectedK8sVersion := range k.k3skubernetesVersions {
+		require.Contains(k.T(), releasedK8sVersions, expectedK8sVersion)
+	}
+}
+
+func (k *KdmChecksTestSuite) TestProvisioningSingleNodeK3SClusters() {
+	require.GreaterOrEqual(k.T(), len(k.providers), 1)
+
+	subSession := k.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := k.client.WithSession(subSession)
+	require.NoError(k.T(), err)
+
+	kubeProvisioningClient, err := k.client.GetKubeAPIProvisioningClient()
+	require.NoError(k.T(), err)
+
+	providerName := k.providers[0]
+	provider := CreateProvider(providerName)
+	nodeRoles := k.nodesAndRoles
+
+	clusterNames := []string{}
+	clusterResps := []*provisioningV1.SteveAPIObject{}
+	k8sVersions := []string{}
+
+	// provisioning k3s clusters using all released k8s versions
```

Comment:
Is this comment needed? As you're already using `logrus` for logging, seems a bit redundant.

Reply:
yeah right, i removed it.